test/e2e/apimachinery/crd_validation_rules.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/features"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+var _ = SIGDescribe("CustomResourceValidationRules [Privileged:ClusterAdmin]", func() {
+	f := framework.NewDefaultFramework("crd-validation-expressions")
+
+	testCrd, err := setupCRD(f, schemaWithValidationExpression, "common-group", "v1")
+	if err != nil {
+		framework.Failf("%v", err)
+	}
+	defer func() {
+		if err := cleanupCRD(f, testCrd); err != nil {
+			framework.Failf("%v", err)
+		}
+	}()
+	crd := testCrd.Crd
+
+	config, err := framework.LoadConfig()
+	framework.ExpectNoError(err, "loading config")
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	framework.ExpectNoError(err, "initializing dynamic client")
+	gvr := schema.GroupVersionResource{
+		Group:    crd.Spec.Group,
+		Version:  crd.Spec.Versions[0].Name,
+		Resource: crd.Spec.Names.Plural,
+	}
+
+	ginkgo.It("MUST fail creation of a custom resource definition with x-kubernetes-validator rules containing invalid syntax", func() {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.CustomResourceValidationExpressions)
```

Comment:
last time I've checked this doesn't do what I think it does, this runs on the e2e binary, is not able to detect the feature gates on the components ... unless you are setting it somewhere in the e2e framework

Reply:
I've removed the skip and added these tags: `[Alpha][Feature:CustomResourceValidationExpressions]`.